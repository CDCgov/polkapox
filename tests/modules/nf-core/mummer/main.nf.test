nextflow_process {

    name "Test Process MUMMER"
    script "modules/nf-core/mummer/main.nf"
    process "MUMMER"

    test("Mummer Test") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test_sub'],
                    [
                        file("${projectDir}/tests/nf-test-data/test_sub.assembly_asm.fasta", checkIfExists: true)
                    ], 
                    [
                        file("${projectDir}/tests/nf-test-data/test_sub.final.fa", checkIfExists: true)
                    ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
