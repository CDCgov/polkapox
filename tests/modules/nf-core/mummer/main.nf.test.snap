{
    "Mummer Test": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "summary": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-07T14:21:42.922476006"
    }
}